renderdoc/driver/metal/metal_types.cpp

```diff
@@ -49,16 +50,19 @@ void DoSerialiseViaResourceId(SerialiserType &ser, type &el)
 
   DoSerialise(ser, id);
 
-  if(ser.IsReading() && rm && !IsStructuredExporting(rm->GetState()))
+  if(ser.IsReading())
   {
     el = NULL;
```

Comment:
Found this problem when doing the conversion from rdc -> XML. 
A serialized NS::String* was not set to NULL causing a crash.

Reply:
Why was an `NS::String*` going through this path being serialised as an Id?